### ISSUE TITLE
[css-overflow-5] `:target-current` pseudo class should also imply `aria-current=true` on an active scroll marker

### ISSUE THREAD

**Issue by w3cbot** [w3c/a11y-tracking#247]
**This is a tracker issue.** Only discuss things here if they are a11y group internal meta-discussions about the issue. **Contribute to the actual discussion at the following link:**

§ https://github.com/w3c/csswg-drafts/issues/12659

**Comment by matatk** [comment]
A couple of bits of info and discussion recently in APA:

* [minutes from our discussion about it](https://www.w3.org/2026/05/13-apa-minutes.html#aaa8): concerns included:
  - are they proposing to set DOM properties or actual ARIA attrs?
  - overreach of this approach
  - inconsistency with other parts of the platform
* Corresponding ARIA issue: https://github.com/w3c/aria/issues/2710
  - Note that this issue also discusses the development of the [CSS-AAM](https://github.com/w3c/css-aam/)

**Comment by AutoSponge** [comment]
daniil responded in the last meeting to say they were only moving forward with aam mapping.

**Comment by matatk** [comment]
> daniil responded in the last meeting to say they were only moving forward with aam mapping.

[Discussed 2026-05-20](https://www.w3.org/2026/05/20-apa-minutes.html#3332)

We think this change allays @stevefaulkner's concern about round-tripping via setting the ARIA _attributes_ rather than DOM/accTree properties.

However we still have the concern as to how the scroll marker may be programmatically set.

*(I've assigned @AutoSponge, @stevefaulkner, and me, as we all expressed an interest on recent APA calls - feel free to remove yourself if you don't want further updates.)*

**Comment by bramus** [comment]
I’m trying to wrap my head around the status of things here, but I can’t seem to figure it out.

Looking up relevant discussions in standards groups I find a lot of issues spread across various venues, but these all just point to each other:

- https://github.com/w3c/csswg-drafts/issues/12659
- https://github.com/w3c/css-aam/issues/15
- https://github.com/w3c/aria/issues/2710
- https://github.com/w3c/a11y-tracking/issues/247

There seems to be [agreement to get this defined in CSS-AAM](https://github.com/w3c/csswg-drafts/issues/12659#issuecomment-4443404384), but [this later comment _(in this very issue)_](https://github.com/w3c/a11y-tracking/issues/247#issuecomment-4503062179) links to [an APA telcon discussion](https://www.w3.org/2026/05/20-apa-minutes.html#3332) that seems to have brought up round-tripping concerns.

What needs to be done to get things rolling here, so that we can [address the original complaint about this _(filed by Sara Soueidan)_](https://issues.chromium.org/issues/439294790)?

Does this round tripping need further discussion? (If so: at which venue?) Or is it just a matter of getting the spec text ready? Or …?

If it helps: There is [a code change](https://chromium-review.googlesource.com/6853905) available in Chromium _(behind a flag)_ that sets `aria-current=true` to `:target-current` as proposed. Maybe this helps to test/verify some things?

**Comment by AutoSponge** [comment]
@matatk @stevefaulkner I was of the understanding that we WANTED an AAM mapping and NOT any type of forced `aria-*`. Keeping `aria-*` explicit and declarative seemed like a consistent position. ~If I'm wrong (and the Chromium code is ideal), please speak up.~

EDIT: based on @bramus clarification below, the Chromium code may be what we want. We would need to make sure WCAG has a new technique article, a warning against mixing techniques (don't use `aria-current` while also using `:target-current` unless they're purposefully redundant and on the same element.

**Comment by bramus** [comment]
Maybe one thing to clarify: I originally wrote _“syncs `aria-current` to `:target-current`”_ in my reply here but the word _“sync”_ is probably ill-chosen as it might imply that `aria-current=true` gets written in the DOM, which is not the case.

Looking at [the code](https://chromium-review.git.corp.google.com/c/chromium/src/+/6853905) (the file `third_party/blink/renderer/modules/accessibility/ax_node_object.cc` specifically), the change this code introduces is that it reports `true` for the `ariaCurrentState` (in the AX tree) when the element is the `:target-current` element.

**Comment by bramus** [comment]
Reading @AutoSponge’s update in https://github.com/w3c/a11y-tracking/issues/247#issuecomment-5403191756 after my clarification, it looks like we’re in agreement.

Practically speaking: how do we continue here? Does this need to be Agenda+’d somewhere to get an official resolution? If so, at which venue?

_(Thanks for bearing with me here. I am trying to understand what is preventing this necessary fix from shipping and am trying to figure out what needs to be done to unblock it)_

**Comment by bramus** [comment]
I was pointed to https://github.com/w3c/aria/issues/2710#issuecomment-3819696897 which has the resolution, so there https://github.com/w3c/css-aam/issues/15 should be able to move forward :)
